tools/debug.py

```diff
@@ -128,57 +128,56 @@ def derive_full_bit_qwords(self, n):
-        else:
-            raise NotImplemented
+        raise NotImplemented
 
     def make_packed_matrix(self, qmat):
         if self.arrangement == 'Sequential':
             raise NotImplemented
         if self.arrangement == 'BitInterleaving':
             raise NotImplemented
-        elif self.arrangement == 'WordInterleaving':
-            qmat = np.zeros([self.row, self.col], dtype=np.uint32)
-
-            orig_col = self.orig.shape[1]
-            full_bit_col = orig_col // self.nbits_per_word
-            residual_bit_col = orig_col % self.nbits_per_word
-
-            def pack_bits(x, num_data, row, col_offset):
-                out = np.zeros([self.nbits], dtype=np.uint32)
-
-                for bi in reversed(range(0, num_data)):
-                    data = x[row, col_offset + bi]
-                    for bc in range(0, self.nbits):
-                        bit = data & 0x1
-                        out[bc] |= (bit << bi)
-                        data = data >> 1
-                return out
-
-            for hi in range(0, self.row):
-                in_col_offset = 0
-                out_col_offset = 0
-
-                for c in range(0, full_bit_col):
-                    out = pack_bits(self.orig, self.nbits_per_word, hi, in_col_offset)
-                    for x in range(self.nbits):
-                        print(f"{c*self.nbits+x}: {format(out[x], '032b')}")
-
-                    import ipdb
-                    ipdb.set_trace()
-
-                    qmat[hi, out_col_offset:out_col_offset + self.nbits] = out
-                    in_col_offset += self.nbits_per_word
-                    out_col_offset += self.nbits
-
-                if residual_bit_col != 0:
-                    out = pack_bits(self.orig, residual_bit_col, hi, in_col_offset)
-                    qmat[hi, out_col_offset:out_col_offset + self.nbits] = out
-        else:
+        if self.arrangement != 'WordInterleaving':
             raise NotImplemented
 
+        qmat = np.zeros([self.row, self.col], dtype=np.uint32)
+
+        orig_col = self.orig.shape[1]
+        full_bit_col = orig_col // self.nbits_per_word
+        residual_bit_col = orig_col % self.nbits_per_word
+
+        def pack_bits(x, num_data, row, col_offset):
+            out = np.zeros([self.nbits], dtype=np.uint32)
+
+            for bi in reversed(range(0, num_data)):
+                data = x[row, col_offset + bi]
+                for bc in range(0, self.nbits):
+                    bit = data & 0x1
+                    out[bc] |= (bit << bi)
+                    data = data >> 1
+            return out
+
+        for hi in range(0, self.row):
+            in_col_offset = 0
+            out_col_offset = 0
+
+            for c in range(0, full_bit_col):
+                out = pack_bits(self.orig, self.nbits_per_word, hi, in_col_offset)
+                for x in range(self.nbits):
+                    print(f"{c*self.nbits+x}: {format(out[x], '032b')}")
+
+                import ipdb
```

Comment:
can we move import at the top of the file? And I'm not sure if we should set_trace every loop.

Reply:
I think it's better to do that too. Can I send it to another PR?

Reply:
fixed.